source/MapPlanetCard.cpp

```diff
@@ -0,0 +1,276 @@
+	reputationLabel = !hasSpaceport ? "No Spaceport" :
+		GameData::GetPolitics().HasDominated(planet) ? "Dominated" :
+		planet->GetGovernment()->IsEnemy() ? "Hostile" :
+		planet->CanLand() ? "Friendly" : "Restricted";
+	
+	sprite = object.GetSprite();
+
+	static const Interface* planetCardInterface = GameData::Interfaces().Get("map planet card");
+	static const float planetIconMaxSize = static_cast<float>(planetCardInterface->GetValue("planet icon max size"));
+	spriteScale = min(.5f, min((planetIconMaxSize) / sprite->Width(), (planetIconMaxSize) / sprite->Height()));
+}
+
+
+
+MapPlanetCard::~MapPlanetCard()
+{
+	auto begin = cards.begin();
+	bool found = false;
+	unsigned i = 0;
+	for(auto &it : cards)
+	{
+		// The ones after the one we deleted need their number adjusted.
+		if(found)
+			--it->number;
+		else if(it == this)
+			found = true;
+		else
+			++i;
+	}
+	// We should be in it, but just to be sure.
+	if(found)
+		cards.erase(begin + i);
+}
+
+
+
+MapPlanetCard::ClickAction MapPlanetCard::Click(int x, int y, int clicks)
+{
+	ClickAction clickAction = ClickAction::NONE;
+	// The isShown variable should have already updated by the drawing of this item.
+	if(isShown) {
+		static const Interface *planetCardInterface = GameData::Interfaces().Get("map planet card");
+		static const double textStart = planetCardInterface->GetValue("text start");
+		static const double categorySize = planetCardInterface->GetValue("category size");
+		static const double categories = planetCardInterface->GetValue("categories");
+		static const double planetIconMaxSize = planetCardInterface->GetValue("planet icon max size");
+
+		// The yCoordinate refers to the center of this object.
+		double relativeY = y - yCoordinate;
+		if(relativeY > 0. && relativeY < AvailableSpace())
+		{
+			isSelected = true;
+
+			// The first category is the planet name and is not selectable.
+			if(x > Screen::Left() + planetIconMaxSize && 
+					relativeY > textStart + categorySize && relativeY < textStart + categorySize * categories)
+				selectedCategory = (relativeY - textStart - categorySize) / categorySize;
+			else
+				clickAction = ClickAction::SELECTED;
+
+			static const int SHOW[4] = {MapPanel::SHOW_REPUTATION, MapPanel::SHOW_SHIPYARD,
+										MapPanel::SHOW_OUTFITTER, MapPanel::SHOW_GOVERNMENT};
+			if(clickAction != ClickAction::SELECTED)
+			{
+				clickAction = static_cast<ClickAction>(SHOW[selectedCategory]);
+				if(clickAction == ClickAction::SHOW_SHIPYARD && clicks > 1)
+					clickAction = ClickAction::GOTO_SHIPYARD;
+				else if(clickAction == ClickAction::SHOW_OUTFITTER && clicks > 1)
+					clickAction = ClickAction::GOTO_OUTFITTER;
+			}
+		}
+	}
+	isSelected = clickAction != ClickAction::NONE;
+	return clickAction;
+}
+
+
+
+bool MapPlanetCard::DrawIfFits(const Point &uiPoint)
+{
+	// Need to update this before checking if the element fits.
+	yCoordinate = uiPoint.Y();
+	isShown = Shown();
+	if(isShown)
+	{
+		static const Font &font = FontSet::Get(14);
```

Comment:
Okay, I wasn't aware of those plans for the future
Yeah that's why I made them static, caus that way we don't reload them all of the time, will change

Reply:
You really want me to make the interface data members? 
Each MapPlanetCard has his own? Because this way currently there are only 4 total variables (and static const so they don't get reinstated every time we reload the planetCards)